agent/engine/docker_client.go

```diff
@@ -125,31 +125,32 @@ type DockerClient interface {
 	ImportLocalEmptyVolumeImage() DockerContainerMetadata
 
 	// CreateContainer creates a container with the provided docker.Config, docker.HostConfig, and name. A timeout value
-	// should be provided for the request.
-	CreateContainer(*docker.Config, *docker.HostConfig, string, time.Duration) DockerContainerMetadata
+	// and a context should be provided for the request.
+	CreateContainer(*docker.Config, *docker.HostConfig, string, time.Duration, context.Context) DockerContainerMetadata
 
-	// StartContainer starts the container identified by the name provided. A timeout value should be provided for the
-	// request.
-	StartContainer(string, time.Duration) DockerContainerMetadata
+	// StartContainer starts the container identified by the name provided. A timeout value and a context should be
+	// provided for the request.
+	StartContainer(string, time.Duration, context.Context) DockerContainerMetadata
```

Comment:
Actually, this is all repetitive as far as I can tell. Lets get rid of time.Duration in all of the signatures.

Reply:
Please see the comments above.